resources/js/menu.js

```diff
@@ -13,65 +13,70 @@
  *  See the License for the specific language governing permissions and
  *  limitations under the License.
  */
+const preferences = require('./preferences.js')
+
+// Save the preferencesMenu then reuse it later on.
+// This way we can use the same menu in the GUI and on the tray
+var preferencesMenu = {
+  label: 'Preferences',
+  submenu: [{
+    label: 'Notification Preferences',
+    submenu: [
+      { label: "Show Notifications", type: 'checkbox', checked: preferences.showNotifications(), click() {
+        preferences.toggleShowNotifications() }
+      },
+      { label: "Play Notification Sound", type: 'checkbox', checked: preferences.notificationSounds(), click() {
+        preferences.toggleNotificationSounds() }
+      },
+      { type: 'separator' },
+      { label: "Display Sender in Notification", type: 'checkbox', checked: preferences.notificationSenderPreviews(), click() {
+        preferences.toggleNotificationSenderPreviews() }
+      },
+      { label: "Display Message Preview in Notification", type: 'checkbox', checked: preferences.notificationMessagePreviews(), click() {
+        preferences.toggleNotificationMessagePreviews() }
+      },
+      { type: 'separator' },
+      { label: "Snooze Desktop Notifications", submenu: [
+        { label: "30 mins", type: 'checkbox', checked: preferences.isSnoozeActive() && preferences.currentSnoozeSelection() == "30_mins", click() {
+          preferences.snooze("30_mins") }
+        },
+        { label: "1 hour", type: 'checkbox', checked: preferences.isSnoozeActive() && preferences.currentSnoozeSelection() == "1_hour", click() {
+          preferences.snooze("1_hour") }
+        },
+        { label: "3 hours", type: 'checkbox', checked: preferences.isSnoozeActive() && preferences.currentSnoozeSelection() == "3_hours", click() {
+          preferences.snooze("3_hours") }
+        },
+        { label: "12 hours", type: 'checkbox', checked: preferences.isSnoozeActive() && preferences.currentSnoozeSelection() == "12_hours", click() {
+          preferences.snooze("12_hours") }
+        }
+      ] }
+    ]
+  }, { type: 'separator' }, {
```

Comment:
I would remove this preference section and put it down below, in the `buildMenu` function. Let's change the preferences in the menu bar to just be the "Notification Preferences".